sequencing-server/src/app.ts

```diff
@@ -902,16 +904,36 @@ app.post('/bulk-get-seqjson-for-seqid-and-simulation-dataset', async (req, res,
  */
 app.post('/get-edsl-for-seqjson', async (req, res, next) => {
   const seqJson = req.body.input.seqJson as SeqJson;
+  const validate = new Ajv({ strict: false }).compile(schema);
+
+  if (!validate(Sequence.fromSeqJson(seqJson))) {
+    throw new Error(
+      `POST /bulk-get-edsl-for-seqjson: Uploaded sequence JSON is invalid and does not match the current spec`,
+    );
+  }
 
   res.json(Sequence.fromSeqJson(seqJson).toEDSLString());
+
   return next();
 });
 
 // Generate Sequence EDSL from many sequence JSONs
 app.post('/bulk-get-edsl-for-seqjson', async (req, res, next) => {
   const seqJsons = req.body.input.seqJsons as SeqJson[];
+  const validate = new Ajv({ strict: false }).compile(schema);
+
+  res.json(
+    seqJsons.map(seqJson => {
+      if (!validate(Sequence.fromSeqJson(seqJson))) {
+        throw new Error(
+          `POST /bulk-get-edsl-for-seqjson: Uploaded sequence JSON is invalid and does not match the current spec`,
+        );
+      }
+
+      return Sequence.fromSeqJson(seqJson).toEDSLString();
```

Comment:
Thoughts on having the validation done inside Sequence.fromSeqJson so we are sure to validate any usage? Are we using that in cases where we might want best-effort conversion and not validate?

Reply:
I thought about putting it inside `fromSeqJson` but I was worried about introducing `Ajv` and the schema inside the preface. I suppose we could pass those in if we wanted to validate inside that function though.

Any thoughts @goetzrrGit ?